src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -112,7 +124,7 @@ static void purgeExecutors() {
      * @return true if the policy was successfully enabled 
      */
     public static boolean tryEnableCancelPolicy(ScheduledExecutorService exec) {
-        if (!PURGE_FORCE) {
+        if (!PURGE_FORCE && (!IS_ANDROID || IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY)) {
```

Comment:
Sorry my boolean logic is all screwed up because I have my own version of this in my head.

My point was and remain, both of these are constant booleans which means whatever logic you want to apply to them can be done once. This yields a single static field which encapsulates both conditions (not Android or API 22+).